src/main/java/us/abstracta/wiresham/Flow.java

```diff
@@ -272,8 +274,16 @@ public int getPortCount() {
 
   public List<Integer> getPorts() {
     return steps.stream()
-        .filter(p -> p instanceof SendPacketStep)
-        .map(PacketStep::getPort)
+        .map(packetStep -> {
+          if (packetStep instanceof SendPacketStep) {
+            return Collections.singletonList(packetStep.getPort());
+          } else if (packetStep instanceof ParallelPacketStep) {
+            return ((ParallelPacketStep) packetStep).getPorts();
+          }
+          return null;
+        })
+        .filter(Objects::nonNull)
```

Comment:
you can avoid this returning Collections.empty() instead. In general is a good practice for collections to return their empty version instead of returning null.